runtime/src/main/java/org/corfudb/runtime/exceptions/StreamingException.java

```diff
@@ -13,11 +15,22 @@
  */
 public class StreamingException extends RuntimeException {
 
-    public StreamingException(Throwable throwable) {
-        super(throwable);
+    public enum ExceptionCause {
+        SUBSCRIBE_ERROR,
```

Comment:
Previously we don't differentiate the cause of StreamingException, and in some places if it's thrown just because it's already subscribed, we might want to no-op so we need some way to tell the cause of the exception.